compiler/src/dotty/tools/dotc/transform/PostTyper.scala

```diff
@@ -172,7 +172,10 @@ class PostTyper extends MacroTransform with InfoTransformer { thisPhase =>
             if sym.is(Param) then
               sym.keepAnnotationsCarrying(thisPhase, Set(defn.ParamMetaAnnot), orNoneOf = defn.NonBeanMetaAnnots)
             else if sym.is(ParamAccessor) then
+              // @publicInBinary is not a meta-annotation and therefore not kept by `keepAnnotationsCarrying`
+              val publicInBinaryAnnotOpt = sym.getAnnotation(defn.PublicInBinaryAnnot)
               sym.keepAnnotationsCarrying(thisPhase, Set(defn.GetterMetaAnnot, defn.FieldMetaAnnot))
```

Comment:
Should we create a `scala.annotation.meta.publicInBinary` meta-annotation and work internally with that one?

Reply:
Could we annotate the definition of `publicInBinary` with `scala.annotation.meta.getter` instead?

Reply:
That did not work. We need to distinguish this annotation after postyper from just `getter`.